### PR TITLE
Add `ResponseCode` output field

### PR DESCRIPTION
**Pull request by necaris**
Adds the HTTP response code to the outputs, to aid in debugging API calls during development.

### CODE REVIEW

**Review by dirien — COMMENTED** [review]
Hi @necaris,

thanks for your contribution! Can you keep the old properties please and mark them as deprecated so we can slowly delete them and avoid any harsh braking changes!

Also I would ask to split the PR into two: Add new properties and return value for better debugging!

Thanks!

**Comment by necaris** [comment]
@dirien thank you for the feedback! I've kept the response code field in this PR and moved the renames to #79 , as suggested!

File: provider/pkg/provider/purrl/purrl.go [hunk 1]
```diff
@@ -55,9 +56,9 @@ func boolPtr(b bool) *bool {
 }
 
 func callAPIEndpoint(method, url, body *string, responseCode *[]string, headers *map[string]string,
-	insecureSkipVerify *bool, caCert, cert, key *string) (*string, error) {
+	insecureSkipVerify *bool, caCert, cert, key *string) (*string, *string, error) {
```
Comment (dirien) [inline provider/pkg/provider/purrl/purrl.go:59]: Can you change the first return type to `*int`

File: provider/pkg/provider/purrl/purrl.go [hunk 2]
```diff
@@ -115,12 +116,12 @@ func callAPIEndpoint(method, url, body *string, responseCode *[]string, headers
...
 	resp, err := client.Do(request)
 	if err != nil {
-		return nil, fmt.Errorf("error calling API endpoint: %v", err)
+		return strPtr(string(resp.StatusCode)), nil, fmt.Errorf("error calling API endpoint: %v", err)
 	}
 	defer resp.Body.Close()
 	respBody, readErr := io.ReadAll(resp.Body)
 	if readErr != nil {
-		return nil, fmt.Errorf("error reading response body: %v", readErr)
+		return strPtr(string(resp.StatusCode)), nil, fmt.Errorf("error reading response body: %v", readErr)
```
Comment (dirien) [inline provider/pkg/provider/purrl/purrl.go:124]: Create a new function:

```go
func intPtr(i int) *int {
	return &i
}
```

and change to `return intPtr(resp.StatusCode), ...

File: provider/pkg/provider/purrl/purrl.go [hunk 3]
```diff
@@ -115,12 +116,12 @@ func callAPIEndpoint(method, url, body *string, responseCode *[]string, headers
 
 	resp, err := client.Do(request)
 	if err != nil {
-		return nil, fmt.Errorf("error calling API endpoint: %v", err)
+		return strPtr(string(resp.StatusCode)), nil, fmt.Errorf("error calling API endpoint: %v", err)
```
Comment (dirien) [inline provider/pkg/provider/purrl/purrl.go:119]: Create a new function:

```go
func intPtr(i int) *int {
	return &i
}
```

and change to `return intPtr(resp.StatusCode), ...

File: provider/pkg/provider/purrl/purrl.go [hunk 4]
```diff
@@ -131,10 +132,10 @@ func callAPIEndpoint(method, url, body *string, responseCode *[]string, headers
 	}
 
 	if !responseCodeChecker(stringConversionList, code) {
-		return nil, errors.New("response code not in list of expected response codes")
+		return strPtr(string(resp.StatusCode)), nil, errors.New("response code not in list of expected response codes")
```
Comment (dirien) [inline provider/pkg/provider/purrl/purrl.go:135]: Create a new function:

```go
func intPtr(i int) *int {
	return &i
}
```

and change to `return intPtr(resp.StatusCode), ...

File: provider/pkg/provider/purrl/purrl.go [hunk 5]
```diff
@@ -131,10 +132,10 @@ func callAPIEndpoint(method, url, body *string, responseCode *[]string, headers
...
 	}
 
-	return strPtr(string(respBody)), nil
+	return strPtr(string(resp.StatusCode)), strPtr(string(respBody)), nil
```
Comment (dirien) [inline provider/pkg/provider/purrl/purrl.go:138]: Create a new function:

```go
func intPtr(i int) *int {
	return &i
}
```

and change to `return intPtr(resp.StatusCode), ...

**Review by dirien — CHANGES_REQUESTED** [review]
Hi @necaris,

thanks for the contribution.

I left some comments for changes.

**Comment by necaris** [comment]
@dirien thank you for the review! Changed the first return value to an int as suggested.

**Review by dirien — APPROVED** [review]
LGTM! Many thanks @necaris for the contributon!

**Comment by dirien** [comment]
And it is released! https://github.com/pulumiverse/pulumi-purrl/releases/tag/v0.4.0

**Pull request merged**